services/oldsService.js

```diff
@@ -62,6 +62,25 @@ const sqlGems = (workerId, start, end) => `
     ORDER BY jobID;
 `;
 
+
+const sqlIntervals = (workerId, start, end) => `
+    SELECT
+        t.jobID,
+        t.startDTS,
+        t.endDTS,
+        (UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS)) as secs,
+        ROUND((UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS))/3600,2) as hours,
+            CONCAT(
+                LPAD((UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS)) DIV 3600 % 60,2, 0),
+                ':',
+                LPAD((UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS)) % 3600 DIV 60, 2, 0),
+                ':00'
+                ) as time
+    FROM TIMINGS t
+    WHERE t.workerID=${workerId} AND t.startDTS>='${start}' AND t.startDTS<'${end}'
```

Comment:
Here you use `t.startDTS<'${end}'` and later in the code you always pass `23:59:59` time here, so if the event happens at 23:59:59 you'll never select it. Use '<=' here or pass next day 00:00:00.